certbot/certbot/_internal/main.py

```diff
@@ -1774,15 +1772,23 @@ def reconfigure(config: configuration.NamespaceConfig,
 
     # this is where lineage_config gets fully filled out (e.g. --apache will set auth and installer)
     installer, auth = plug_sel.choose_configurator_plugins(lineage_config, plugins, "certonly")
-    le_client = _init_le_client(lineage_config, auth, installer)
+
+    # make a deep copy of lineage_config because we're about to modify it for a test dry run
+    dry_run_lineage_config = copy.deepcopy(lineage_config)
+
+    dry_run_lineage_config.dry_run = True
```

Comment:
Trying to think of who this wouldn't work for...we're editing the config file, so even if they're running `certonly` instead of `renew`, it's reading the values from the file, not using `duplicate`, that's the whole point. In which case all values should be set already, and nothing should have to be read from input. I worry a little that I just haven't thought of some case. But I guess the failure there is that it doesn't for some weird case, in which case we can fix it if we get complaints, vs. what you said, about the general assurance that things we change going forward don't mess things up. That sounds good to me.